### PR TITLE
Fix server Startup discovery

### LINKED ISSUE #1789: server discovery broken in master?
### Describe the bug

If i compile from `release-0.13.z` server discovery works against 10.8.
If i compile from `master` server discovery don't work.

### Logs

_No response_

### Application version

master

### Where did you install the app from?

Sideloaded APK

### Device information

Z9X

### Android version

Android 9

### Jellyfin server version

10.8

### PR DESCRIPTION
**Pull request by Andy2244**
**Changes**
loadDiscoveryServers() was never called
fix binding.discoveryServers in wrong scope

**Issues**
Fixes #1789

### CODE REVIEW

File: app/src/main/java/org/jellyfin/androidtv/ui/startup/StartupViewModel.kt [hunk 1]
```diff
@@ -2,10 +2,13 @@ package org.jellyfin.androidtv.ui.startup
 
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.coroutineScope
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/ui/startup/StartupViewModel.kt:11]: You added some imports that don't appear the be used.

File: app/src/main/java/org/jellyfin/androidtv/auth/repository/ServerRepository.kt [hunk 2]
```diff
@@ -66,15 +68,17 @@ class ServerRepositoryImpl(
...
-		val servers = mutableListOf<Server>()
-
-		jellyfin.discovery
-			.discoverLocalServers()
-			.map(ServerDiscoveryInfo::toServer)
-			.collect { server ->
-				servers += server
-				_discoveredServers.emit(servers.toList())
-			}
+		withContext(Dispatchers.IO) {
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/auth/repository/ServerRepository.kt:71]: Use `viewModelScope.launch(Dispatchers.IO)` in the StartupViewModel instead.

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/jellyfin/androidtv/auth/repository/ServerRepository.kt (hunk 2 above)
Comment (Andy2244) [reply app/src/main/java/org/jellyfin/androidtv/auth/repository/ServerRepository.kt:71]: um i have no clue how to get `startupViewModel` from the ServerRepository scope?

Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/androidtv/auth/repository/ServerRepository.kt:71]: No I mean to replace line 73 in the viewmodel (`viewModelScope.launch { serverRepository.loadDiscoveryServers() }`) with `viewModelScope.launch(Dispatchers.IO) { serverRepository.loadDiscoveryServers() }` so you don't need the `withContext(){}` here.

Comment (Andy2244) [reply app/src/main/java/org/jellyfin/androidtv/auth/repository/ServerRepository.kt:71]: ah lol, yeah makes more sense.

**Comment by Andy2244** [comment]
ok fixed

File: app/src/main/java/org/jellyfin/androidtv/auth/repository/ServerRepository.kt [hunk 3]
```diff
@@ -1,12 +1,14 @@
 package org.jellyfin.androidtv.auth.repository
 
+import kotlinx.coroutines.Dispatchers
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/auth/repository/ServerRepository.kt:3]: Unused import

File: app/src/main/java/org/jellyfin/androidtv/auth/repository/ServerRepository.kt [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.firstOrNull
 import kotlinx.coroutines.flow.flow
 import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.withContext
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/auth/repository/ServerRepository.kt:11]: Unused import

File: app/src/main/java/org/jellyfin/androidtv/ui/startup/StartupViewModel.kt (hunk 1 above)
Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/androidtv/ui/startup/StartupViewModel.kt:11]: Unused import

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

**Pull request merged**